src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractExtensionMethodImportCompletionProvider.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion.Log;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers
+{
+    internal abstract class AbstractExtensionMethodImportCompletionProvider : AbstractImportCompletionProvider
+    {
+        protected abstract bool TryGetReceiverTypeSymbol(SyntaxContext syntaxContext, [NotNullWhen(true)] out ITypeSymbol? receiverTypeSymbol);
+
+        protected override bool ShouldProvideCompletion(Document document, SyntaxContext syntaxContext)
+            => syntaxContext.IsRightOfNameSeparator && IsAddingImportsSupported(document);
+
+        protected async override Task AddCompletionItemsAsync(
+            CompletionContext completionContext,
+            SyntaxContext syntaxContext,
+            HashSet<string> namespaceInScope,
+            bool isExpandedCompletion,
+            CancellationToken cancellationToken)
+        {
+            if (TryGetReceiverTypeSymbol(syntaxContext, out var receiverTypeSymbol))
+            {
+                using var telemetryCounter = new TelemetryCounter();
+                var ticks = Environment.TickCount;
+
+                var project = completionContext.Document.Project;
+
+                using var allTypeNamesDisposer = PooledHashSet<string>.GetInstance(out var allTypeNames);
+                allTypeNames.Add(receiverTypeSymbol.MetadataName);
+                allTypeNames.AddRange(receiverTypeSymbol.GetBaseTypes().Concat(receiverTypeSymbol.GetAllInterfacesIncludingThis()).Select(s => s.MetadataName));
+
+                // interface doesn't inherit from object, but is implicitly convertable to object type.
```

Comment:
everything is implicitly convertible to object (except maybe pointers).  Not sure why this is needed.

Reply:
Because neither `GetBaseTypes` or `GetAllInterfacesIncludingThis` return Object for interfaces, and we would miss those extension methods for object if receiver symbol is an interface

Reply:
my point is: we should always add `Object` since everything derives from object.

Reply:
Ah, I see. It would be already in the hashset for other cases, but sure.